File: protenium-iro-hub/app/src/main/java/com/protenium/irohub/ui/start_up/splash/SplashActivity.java
package com.protenium.irohub.ui.start_up.splash;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.protenium.irohub.R;
import com.protenium.irohub.base.BaseActivity;
import com.protenium.irohub.ui.start_up.auth.activity.AuthActivity;

public class SplashActivity extends BaseActivity {

    Handler handler;

    @Override
    public int setLayout() {
        return R.layout.activity_splash;
    }

    @Override
    public boolean setToolbar() {
        return false;
    }

    @Override
    public boolean hideStatusBar() {
        return true;
    }

    @Override
    public boolean setFullScreen() {
        return true;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashActivity.this, AuthActivity.class);
                startActivity(intent);
                finish();
            }
        }, 3000);
    }
}

File: jeetmeet/app/src/main/java/com/iroid/jeetmeet/start_up/splash/SplashActivity.kt
package com.iroid.jeetmeet.start_up.splash

import android.content.Intent
import com.iroid.jeetmeet.R
import com.iroid.jeetmeet.base.BaseActivity
import com.iroid.jeetmeet.databinding.ActivitySplashBinding
import com.iroid.jeetmeet.start_up.auth.activity.AuthActivity
import java.util.*

class SplashActivity : BaseActivity<ActivitySplashBinding>()
{
    override val layoutId: Int
        get() = R.layout.activity_splash
    override val setToolbar: Boolean
        get() = false
    override val hideStatusBar: Boolean
        get() = true

    override fun getViewBinging(): ActivitySplashBinding = ActivitySplashBinding.inflate(layoutInflater)


    override fun initData()
    {
        Timer().schedule(object : TimerTask() {
            override fun run() {
                    startActivity(Intent(this@SplashActivity , AuthActivity::class.java))
                    finish()
            }
        } , 3000)
    }

    override fun fragmentLaunch() {
    }

    override fun setupUI() {
    }

    override fun setupViewModel() {

    }

    override fun setupObserver() {
    }

    override fun onClicks() {
    }

}